# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmGC.h
# Commit: f68854b70d6a
# Full Hash: f68854b70d6a22b76f81341182018d32bad08f8a
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-08-06 15:44:30
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 1: Reserve two slots after stack arguments for the future tls preservation. r=lth
#   
#   We are going to remove Frame::tls and support trampolines for indirect calls, so we need to get rid of using Frame::tls.
#   In this and the followup patches I will iteratively remove all dependencies of Frame::tls and remove it eventually.
#   
# ==============================================================================

diff -r 451369452f15 -r f68854b70d6a js/src/wasm/WasmGC.h
--- a/js/src/wasm/WasmGC.h	Thu Aug 06 04:28:01 2020 +0000
+++ b/js/src/wasm/WasmGC.h	Thu Aug 06 04:57:48 2020 +0000
@@ -252,7 +252,7 @@
 // the complete native-ABI-level call signature.
 template <class T>
 static inline size_t StackArgAreaSizeUnaligned(const T& argTypes) {
-  ABIArgIter<const T> i(argTypes);
+  WasmABIArgIter<const T> i(argTypes);
   while (!i.done()) {
     i++;
   }
@@ -261,7 +261,7 @@
 
 static inline size_t StackArgAreaSizeUnaligned(
     const SymbolicAddressSignature& saSig) {
-  // ABIArgIter::ABIArgIter wants the items to be iterated over to be
+  // WasmABIArgIter::ABIArgIter wants the items to be iterated over to be
   // presented in some type that has methods length() and operator[].  So we
   // have to wrap up |saSig|'s array of types in this API-matching class.
   class MOZ_STACK_CLASS ItemsAndLength {